lib/dangermattic/plugins/labels_checker.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Danger
+  # Plugin to check for labels in a PR.
+  class LabelsChecker < Plugin
+    DEFAULT_DO_NOT_MERGE_LABEL = 'Do Not Merge'
+    DEFAULT_LABELS_WARNING = 'Please add the required labels for this project.'
+    DEFAULT_REQUIRED_LABELS = [/.*\S.*/].freeze
```

Comment:
I'm not sure I'm following why this would make sense as a default set of required label.
 - If I understand this regex correctly, it's gonna match zero-or-more characters (`.*`) followed by a non-whitespace character (`\S`) , followed by zero-or-more characters again. Which basically matches everything (except labels whose name is composed entirely of whitespaces, which is unlikely and probably not allowed by GitHub anyway)
 - Besides, at first I thought the goal of using a "match-all" regex as default would be a derived way to check if "PR has at least one label", which I think should work and could be a good solution, except that:
    - In practice that's not how you use this, since you hardcoded a `if github_labels.empty?` with `warn('PR is missing at least one label.')` below instead
    - And a simpler Regex like `/.+/` should be enough to do that trick anyway I think

Reply:
I think the simpler approach here would be to set `DEFAULT_REQUIRED_LABELS = [/.*/].freeze # require at least one label, without any constraint on which one`, or maybe make `[]` be the default instead? And then remove the special case of `github_labels.empty?` check on line 30, to instead let the more generic check on the required_label to take care of that case.

 - If one wants to use the plugin just for the do_not_merge labels, they can use `[]` for the required labels
 - If one wants to use the plugin for requiring at least one label, without any rule about which label that is, they can use `[/.*/]`
 - If one wants to request a specific set of labels, then one could use `[/^feature:/, /^type:/]`, and then the case of "require at least one label overall" would already be included in the case of "require at least one feature label and one type label" anyway

Reply:
> And then remove the special case of github_labels.empty? check on line 30, to instead let the more generic check on the required_label to take care of that case.

I think that's a better and more general solution (also addressing your comment [here](https://github.com/Automattic/dangermattic/pull/7#discussion_r1262727924)). Having at least one label was the hard requirement on the Peril rule, but I think it is nicer (and a better strategy in general) to make it more flexible if we can so also more repos can adopt these checks. I liked your suggestion, it's a good balance of flexibility and conventions 👍